### ISSUE TITLE
I met a problem that bothered me a lot，It's about validation

### ISSUE THREAD

**Issue by ttttttw** [ajv-validator/ajv#1276]
![image](https://user-images.githubusercontent.com/24540969/92429928-018c1e00-f1c6-11ea-8fa3-245fe24ea93c.png)
My info depends on other schema，
I need to rely on info for rendering。
but，If info exists, it will be validated。
Because this is a logic of dynamic processing, info may exist or may not exist depending on the operation of the page。
So, when I don't need the info, I remove the info。
But my page rendering will rely on info, embarrassing。

In short, I hope to skip the verification of info in some way

**" version": "6.12.4" **

**Comment by epoberezkin** [comment]
Fields are only validated if present, they are not required without "required" keyword. The schema fragment looks incorrect though - please review.

**Comment by litw5034** [comment]
I know what you're talking about. My problem now is that the info field may have content or may not have content, which is determined by page operation. But my page rendering depends on this info field. Now there is a situation that I do not have the specific content of this info field due to page operation, but my info references other schemas. At this time, if the info field exists, the verification will be run. So I had to delete the info and send it to the server. However, my page will retrieve the data stored before by the server next time. At this time, there may be no info field, so the page crashes.
So I wish I had an info field, but in some cases I could skip validation

**Comment by ttttttw** [comment]
> Fields are only validated if present, they are not required without "required" keyword. The schema fragment looks incorrect though - please review.

I know what you're talking about. My problem now is that the info field may have content or may not have content, which is determined by page operation. But my page rendering depends on this info field. Now there is a situation that I do not have the specific content of this info field due to page operation, but my info references other schemas. At this time, if the info field exists, the verification will be run. So I had to delete the info and send it to the server. However, my page will retrieve the data stored before by the server next time. At this time, there may be no info field, so the page crashes.
So I wish I had an info field, but in some cases I could skip validation

**Comment by epoberezkin** [comment]
If you don't want the field validated when it is present, you need a separate schema that does not validate this field (you could modify the schema in your code).

**Comment by ttttttw** [comment]
> If you don't want the field validated when it is present, you need a separate schema that does not validate this field (you could modify the schema in your code).

The way I deal with it now is to check whether the received data contains the info field in the code. If it is, I will skip it. If not, I will supplement it. But I think it is too low to be expanded and customized. I want to find a better way. Do you have any better suggestions?
